framework/mongo/utils.py

```diff
@@ -61,15 +62,78 @@ def wrapper(cls):
     return wrapper
 
 
-def get_or_http_error(Model, pk):
-    instance = Model.load(pk)
-    if getattr(instance, 'is_deleted', False):
-        raise HTTPError(http.GONE, data=dict(
-            message_long="This resource has been deleted"
-        ))
-    if not instance:
-        raise HTTPError(http.NOT_FOUND, data=dict(
-            message_long="No resource with that primary key could be found"
-        ))
-    else:
+def get_or_http_error(Model, pk_or_query):
+    """Load an instance of Model by primary key or modularodm.Q query. Raise an appropriate
+    HTTPError if no record is found or if the query fails to find a unique record
+
+    param Model: <StoredObject> subclass
+    param pk_or_query: either-
+      - a <basestring> representation of the record's primary key, e.g. 'abcdef'
+      - a <QueryBase> subclass query to uniquely select a record, e.g.
+        Q('title', 'eq', 'Entitled') & Q('version', 'eq', 1)
+    return Model instance
+    """
+
+    name = str(Model)
```

Comment:
We probably shouldn't rely on the implementation of `__str__` for displaying resource names to users. There are models that don't match the language we use on the website, e.g. "Pointer" vs. "Links". 

At the very least, this should be a parameter.